numpy/core/code_generators/ufunc_docstrings.py

```diff
@@ -2011,7 +2011,10 @@ def add_newdoc(place, name, doc):
     -----
     Logarithm is a multivalued function: for each `x` there is an infinite
     number of `z` such that `exp(z) = x`. The convention is to return the
-    `z` whose imaginary part lies in `[-pi, pi]`.
+    `z` whose imaginary part lies in `(-pi, pi]`.
+
+    In the cases where x array element input value tends to -1. 
+    The result is in the vicinity of `-pi`, that it evaluates to `-pi`.
```

Comment:
This is fine, but I wonder if we can clean up the grammar a bit and make it easier to read.  Just a thought, but a bit of grammar cleanup is also enough:
> Results of ``-pi`` are possible when the exact result would be very close to ``-pi``.  This for example happens for input approaching -1.

Reply:
Actually, I had not read carefully enough.  `-1` is misleading, this occurs for complex numbers with negative real part and very small negative complex part (approaching 0).  Since the branch-cut is at `x.imag == 0` when `x.real < 0`.